XcodeServerSDKTests/RepositoryTests.swift

```diff
@@ -79,5 +79,35 @@ class RepositoryTests: XCTestCase {
         sshEnum = .LoggedInReadWrite
         XCTAssertEqual(sshEnum.toString(), "All logged in users can read and write")
     }
+    
+    // MARK: API Routes tests
+    func testGetRepositories() {
+        let expectation = self.expectationWithDescription("Get Repositories")
+        let server = self.getRecordingXcodeServer("get_repositories")
+        
+        server.getRepositories() { (repositories, error) in
+            XCTAssertNil(error, "Error should be nil")
+            XCTAssertNotNil(repositories, "Repositories shouldn't be nil")
+            
+            if let repos = repositories {
+                XCTAssertEqual(repos.count, 2, "There should be two repositories available")
+                
+                for (index, repo) in repos.enumerate() {
+                    XCTAssertEqual(repo.name, "Test\(index + 1)")
```

Comment:
No, sorry, I didn't explain my point properly. The test is great and this is exactly how we should write them.

My view is that we should *test* the things that the Xcode Server API is promising to always be true. You have two repos right now, so the API should always return two repos. You have two repos with specific names and properties and again, that's exactly what the API should return. 
**BUT**... the order in which it will return these is not promised by the API in any way. So in theory, if you ran this again on a live server, the order in which the API returns the repos might get reversed. Which is completely legal from the API's perspective, but our test would suddenly fail, because `Test2` would come before `Test1`. Which it shouldn't, because the API is still doing the things it promised to do.

So try keep the tests as **specific** (not generic) as possible to the things that will always be true in this state of your Xcode Server. But do not test things that are not guaranteed by the API, like the order in which the repositories are returned.

Another example might be integrations and bots. When you ask for integrations, they are guaranteed to be returned in chronological order. So you might assert that the time/number of each integration, when iterating over them, is always going up or down. Bots, on the other hand, don't have a promised ordering, so you shouldn't test that bot A comes before bot B, just because it happened once.

Does this make sense? Tell me if not and I'll try to reword. This is very important that we're on the same page.

So the only change I'm suggesting is that you change the asserts for the correct names from being order-dependent (`Test1` coming before `Test2`) to being order-independent, such as

```
let repoNames = Set(repos.map { $0.name })
for (index, repo) in repos.enumerate() {
  XCTAssertTrue(repoNames.contains("Test\(index + 1)"))
}
```

This way it doesn't matter the order in which the repos are returned, the test will only make sure that there is a `Test1` and `Test2` repos returned.